lib/std/build.zig

```diff
@@ -258,9 +258,15 @@ pub const Builder = struct {
         }));
     }
 
-    pub fn addSharedLibrary(self: *Builder, name: []const u8, root_src: ?[]const u8, ver: Version) *LibExeObjStep {
+    pub fn addSharedLibrary(self: *Builder, name: []const u8, root_src: ?[]const u8, kind: union(enum) {
+        versioned: Version,
+        unversioned: void,
+    }) *LibExeObjStep {
```

Comment:
This takes a tagged union instead of `?Version` due to @andrewrk's suggestion in https://github.com/ziglang/zig/issues/2230#issuecomment-558870806. However, I'm not sure this achieves the desired effect. Leaving off the version is still the "easier" option, and setting a version becomes more cumbersome:
```zig
const lib = b.addSharedLibrary("foo", "foo.zig", .unversioned);
```
```zig
const lib = b.addSharedLibrary("foo", "foo.zig", .{ .versioned = b.version(0,1,0) });
```

Reply:
What if `b.version()` returned an instance of this union?

That way its: `b.version(0,1,0)` vs `.unversioned`.

Reply:
Good call, seems  obvious now.